vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+  warning = FALSE,
+  message = FALSE,
+  dpi = 300,
+  comment = "#>",
+  out.width = "100%",
+  fig.height = 6,
+  fig.width = 8
+)
+```
+
+## Introduction
+
+In Modeling and Simulation (M&S) workflows, we often need to work with multiple 
+observed and/or simulated datasets. Further, some of these datasets can be linked 
+together in a group (e.g., the simulated and the correnspoding observed data)
+for analysis and visualization.
+
+The `DataCombined` class in `{ospsuite}` provides a container to store 
+data, to group them, and to transform them further. The class accepts data 
+coming from `SimulationResults` or as a `DataSet` object (see [Observed data](observed-data.html)).
+
+Let's first generate some simulation results and load observed data.
+
+```{r}
+library(ospsuite)
+
+# Simulation results
+simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+sim <- loadSimulation(simFilePath)
+simResults <- runSimulation(sim)
+outputPath <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
+
+# observed data
+obsData <- lapply(
+  c("ObsDataAciclovir_1.pkml", "ObsDataAciclovir_2.pkml", "ObsDataAciclovir_3.pkml"),
+  function(x) loadDataSetFromPKML(system.file("extdata", x, package = "ospsuite"))
+)
+# Name the elements of the list according to the names of the loaded `DataSet`.
+names(obsData) <- lapply(obsData, function(x) x$name)
+```
+
+Typically, `DataCombined` is used to create figures using the `plotXXX()` functions. 
+The different plot types are described in [Visualizations with `DataCombined`](data-combined-plotting.html). 
+To visualize the different functionalities of `DataCombined` in this document, 
+we will use the `plotIndividualTimeProfile()` function.
+
+## Creating `DataCombined` object
+
+First, we create a new instance of `DataCombined` class.
+
+```{r}
+myDataCombined <- DataCombined$new()
+```
+
+Then we add simulated results to this object using `$addSimulationResults()`:
+
+```{r}
+myDataCombined$addSimulationResults(
+  simulationResults = simResults,
+  quantitiesOrPaths = outputPath,
+  names = "Aciclovir Plasma",
+  groups = "Aciclovir PVB"
+)
+```
+
+Next we add observed data to this object using `$addDataSets()`:
+
+```{r}
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`, 
+  groups = "Aciclovir PVB"
+)
+```
+
+Every data, be it simulated results or from `DataSet`, must have a unique name 
+within `DataCombined`. If not specified by user, the path of simulated results 
+or the `$name` property of the `DataSet` are used as the name. Alternatively, we 
+can define the name when adding the data, as in the above example adding simulated 
+results.
+
+There are a few things to keep in mind here:
+
+- It doesn't matter in which order observed or simulated datasets are added.
+- You can add multiple `DataSet` objects in `$addDataSets()` method call.
+- You can add only a single instance of `SimulationResults` in `$addSimulationResults()` method call.
```

Comment:
It's unclear to me what happens if e.g. I want to add additional outputs from the same simulation results later. 
E.g. I tried this:
```
> myDataCombined <- DataCombined$new()
> myDataCombined$addSimulationResults(simulationResults = simResults,quantitiesOrPaths = outputPath,names = "Aciclovir Plasma",  groups = "Aciclovir PVB")
> myDataCombined$addSimulationResults(simulationResults = simResults,quantitiesOrPaths = outputPath2,names = "Blood Cells",  groups = "Aciclovir PVB")
> myDataCombined$names
[1] "Aciclovir Plasma" "Blood Cells" 
```
Was it OK and do I have 2 outputs now in my data combined object?

Now I created another simulation results from a different simulation and add it to my DataCombined object:
```
simResults2 <- runSimulation(sim2)
myDataCombined$addSimulationResults(simulationResults = simResults2,quantitiesOrPaths = outputPath,names = "Aciclovir Plasma 3",  groups = "Aciclovir PVB")
```
No error or warning here and my DataCombined object reports 3 simulated results outputs:
```
> myDataCombined$names
[1] "Aciclovir Plasma"   "Aciclovir Plasma 3" "Blood Cells" 
```
So does it really have 3 outputs from 2 different simulation results now or is it in some inconsistent state in fact?

Reply:
It does have 3 outputs now, you did it exactly how it is intended to be. Could you specify what is not clear to you?

Reply:
Probably just a misunderstanding on my side. I thought first, a can call `$addSimulationResults()` only once per data combined - but it's not what the docu says (it says "1 simulation results per call"). I think all good here.